osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/GetFirmwareVersionRequestMessageProcessor.java

```diff
@@ -32,6 +33,7 @@ protected GetFirmwareVersionRequestMessageProcessor() {
     protected Serializable handleMessage(final DlmsConnectionManager conn, final DlmsDevice device,
             final Serializable requestObject) throws OsgpException {
 
-        return (Serializable) this.configurationService.requestFirmwareVersion(conn, device);
+        final GetFirmwareVersionQueryDto queryDto = (GetFirmwareVersionQueryDto) requestObject;
+        return this.configurationService.requestFirmwareVersion(conn, device, queryDto);
```

Comment:
return this.configurationService.requestFirmwareVersion(conn, device, (GetFirmwareVersionQueryDto) requestObject);